fuel-core/src/executor.rs

```diff
@@ -267,87 +284,61 @@ impl Executor {
+                    execution_data,
+                    execution_kind,
+                    &mut tx_db_transaction,
+                );
 
-            let checked_tx: CheckedTransaction = tx
-                .clone()
-                .into_checked_basic(
-                    block_height as Word,
-                    &self.config.chain_conf.transaction_parameters,
-                )?
-                .into();
-
-            match checked_tx {
-                CheckedTransaction::Script(script) => {
-                    *tx = self.execute_create_or_script(
-                        idx,
-                        script,
-                        &block.header,
-                        execution_data,
-                        block_db_transaction,
-                        execution_kind,
-                    )?
-                }
-                CheckedTransaction::Create(create) => {
-                    *tx = self.execute_create_or_script(
-                        idx,
-                        create,
-                        &block.header,
-                        execution_data,
-                        block_db_transaction,
-                        execution_kind,
-                    )?
+                if let Err(err) = result {
+                    match execution_kind {
+                        ExecutionKind::Production => {
+                            // If, during block production, we get an invalid transaction, remove it
+                            // from the block and continue block creation. An invalid transaction
+                            // means that the caller didn't validate it first, so maybe something
+                            // is wrong with validation rules in the `TxPool`(or in another place that
+                            // should validate it). Or we forgot to clean up some dependent/conflict
+                            // transactions. But it definitely means that something went wrong, and
+                            // we must fix it.
+                            execution_data.skipped_transactions.push((tx, err));
+                            return None
+                        }
+                        ExecutionKind::Validation => return Some(Err(err)),
+                    }
                 }
-                CheckedTransaction::Mint(mint) => {
-                    // Right now, we only support `Mint` transactions for coinbase,
-                    // which are processed separately as a first transaction.
-                    //
-                    // All other `Mint` transaction is not allowed.
-                    let (mint, _): (Mint, _) = mint.into();
-                    return Err(Error::NotSupportedTransaction(Box::new(mint.into())))
+                if let Err(err) = tx_db_transaction.commit() {
+                    return Some(Err(err.into()))
                 }
-            };
-        }
+                tx_index += 1;
+                Some(Ok(tx))
+            })
```

Comment:
```suggestion
            })
            .map(|tx| {
               if tx.is_ok() {
                  tx_index += 1
                }
                tx
              })
```
If you do this and remove the above increments then you can avoid incrementing the index in multiple places

Reply:
I need `tx_index` to execute transaction

Reply:
Yeh that's ok as you are taking a copy into the above closure

Reply:
And this will mean the skipped transactions are already filtered out

Reply:
I don't need a copy with the value `0` in the closure above=) I need to update it after each valid transaction=) I can't have at the same time two closures: first to read `tx_index` and second to write into `tx_index`.

Reply:
Yeh I see because of the double borrow issue. You might need to do something like [this](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=ad2c53b5399b7479e8ab7d39860f58ce)

Reply:
I already do that=)

Reply:
I mean make a function for what is within the filter and then check if it's `Some(Ok())` before incrementing

Reply:
I got it, but it is still the same because I have only one place with `tx_index += 1` after `execute_transaction` at the end of the closure.

Added one more inner closure to finish this thread=)